src/uwtools/drivers/driver.py

```diff
@@ -196,6 +165,133 @@ def _resources(self) -> Dict[str, Any]:
         """
         Returns configuration data for the runscript.
         """
+        try:
+            platform = self._config["platform"]
+        except KeyError as e:
+            raise UWConfigError("Required 'platform' block missing in config") from e
+        return {
+            "account": platform["account"],
+            "rundir": self._rundir,
+        }
+
+    @property
+    def _rundir(self) -> Path:
+        """
+        The path to the component's run directory.
+        """
+        return Path(self._driver_config["run_dir"])
+
+    def _taskname(self, suffix: str) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s" % (self._driver_name, suffix)
+
+    def _taskname_with_cycle(self, cycle: datetime, suffix: str) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s %s" % (cycle.strftime("%Y%m%d %HZ"), self._driver_name, suffix)
+
+    def _taskname_with_cycle_and_leadtime(
+        self, cycle: datetime, leadtime: timedelta, suffix: str
+    ) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s %s" % (
+            (cycle + leadtime).strftime("%Y%m%d %H:%M:%S"),
+            self._driver_name,
+            suffix,
+        )
+
+    def _validate(self) -> None:
+        """
+        Perform all necessary schema validation.
+        """
+        for schema_name in (self._driver_name.replace("_", "-"), "platform"):
+            validate_internal(schema_name=schema_name, config=self._config)
+
+
+class Driver(Assets):
+    """
+    An abstract class for standalone component drivers.
+    """
+
+    def __init__(
+        self,
+        config: Optional[Union[dict, Path]],
+        dry_run: bool = False,
+        batch: bool = False,
+        cycle: Optional[datetime] = None,
+        leadtime: Optional[timedelta] = None,
+        key_path: Optional[List[str]] = None,
+    ) -> None:
+        """
+        A standalone component driver.
+
+        :param config: Path to config file (read stdin if missing or None).
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        :param cycle: The cycle.
+        :param key_path: Keys leading through the config to the driver's configuration block.
+        :param leadtime: The leadtime.
+        """
+        super().__init__(
+            config=config,
+            dry_run=dry_run,
+            batch=batch,
+            cycle=cycle,
+            key_path=key_path,
+            leadtime=leadtime,
+        )
```

Comment:
If I'm not mistaken, this method is identical to the one in the superclass except for the docstring, which IMO is not useful enough to justify the duplication. Can it be removed and just inherit the identical (I think) `Assets` constructor?

Reply:
Huh, turns out it can! I can remove that from my habits now!